### PR TITLE
Feat/event lists

### PR DESCRIPTION
**Pull request by lXSPandora**

### CODE REVIEW

File: src/TypeDefinition.js [hunk 1]
```diff
@@ -8,6 +9,6 @@ export type Dataloaders = {
 };
 
 export type GraphQLContext = {
-  user?: Object,
+  user?: UserType,
```
Comment (socksrust) [inline src/TypeDefinition.js:12]: niceyy

File: src/loader/EventLoader.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+type Schedule = {
+  talker: string,
+  title: string,
+  description: string,
+  time: number,
+};
+
+export type EventType = {
+  id: string,
+  _id: string,
```
Comment (socksrust) [inline src/loader/EventLoader.js:18]: this is a ObjectID from 'mongoose'

File: src/loader/EventLoader.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  image: string,
+  title: string,
+  description: string,
+  date: string,
```
Comment (socksrust) [inline src/loader/EventLoader.js:22]: this is a Date

File: src/model/Event.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import mongoose from 'mongoose';
 const SchemaSchedule = new mongoose.Schema({
   talker: {
     type: String,
-    required: true,
+    required: false,
```
Comment (socksrust) [inline src/model/Event.js:7]: :(

File: src/model/Event.js [hunk 5]
```diff
@@ -15,9 +15,8 @@ const SchemaSchedule = new mongoose.Schema({
     required: false,
   },
   time: {
-    type: Number,
-    required: false,
-    min: 0,
+    type: String,
+    required: true,
```
Comment (socksrust) [inline src/model/Event.js:19]: nice

File: src/mutation/AddEventMutation.js [hunk 6]
```diff
@@ -53,7 +60,7 @@ export default mutationWithClientMutationId({
       ),
     },
   },
-  mutateAndGetPayload: async (args, context) => {
+  mutateAndGetPayload: async (args: EventType, context: GraphQLContext) => {
```
Comment (socksrust) [inline src/mutation/AddEventMutation.js:63]: awesomee

File: src/mutation/AddEventMutation.js [hunk 7]
```diff
@@ -65,32 +72,38 @@ export default mutationWithClientMutationId({
...
     });
-    const event = await data.save();
+    await data.save();
 
-    return event;
+    // return event;
+    return {
+      message: 'Event created with success',
+      error: null,
+    };
```
Comment (socksrust) [inline src/mutation/AddEventMutation.js:97]: we need to return the event :p

File: src/mutation/AttendToEventMutation.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    const { publicList, publicLimit, waitList } = event;
+
+    if (publicList.length >= parseInt(publicLimit)) {
+      await event.update({
+        waitList: [...waitList, user._id],
+      });
+      return {
+        message: 'You have been added to the wait list',
```
Comment (socksrust) [inline src/mutation/AttendToEventMutation.js:52]: heeell yeahh

**Review by socksrust — APPROVED** [review]
u totally rocked here, thnx

File: src/model/Event.js (hunk 4 above)
Comment (lXSPandora) [reply src/model/Event.js:7]: this is not a required field

File: src/model/Event.js (hunk 5 above)
Comment (lXSPandora) [reply src/model/Event.js:19]: tks

File: src/mutation/AddEventMutation.js (hunk 7 above)
Comment (lXSPandora) [reply src/mutation/AddEventMutation.js:97]: yes but we need to put at the output fields

File: src/mutation/AttendToEventMutation.js (hunk 8 above)
Comment (lXSPandora) [reply src/mutation/AttendToEventMutation.js:52]: HAHA

File: src/loader/EventLoader.js (hunk 3 above)
Comment (lXSPandora) [reply src/loader/EventLoader.js:22]: its a string on the model

File: src/loader/EventLoader.js (hunk 2 above)
Comment (lXSPandora) [reply src/loader/EventLoader.js:18]: i think we dont have this type i think is fine as a string

File: src/TypeDefinition.js (hunk 1 above)
Comment (lXSPandora) [reply src/TypeDefinition.js:12]: tks

File: src/mutation/AddEventMutation.js (hunk 7 above)
Comment (jgcmarins) [reply src/mutation/AddEventMutation.js:97]: @lXSPandora , @jaburcodes is right.
We should return an Event.
Maybe we should think of a list of events on client side, which represents a connection on server side, so we should return an Event Edge

**Pull request merged**